### DISCUSSION TITLE
Wireless-Tag WT32S3-86V

### DISCUSSION THREAD

**Discussion by htvekov** [HASwitchPlate/openHASP#435] [Hardware]
![IMG20230325125248](https://user-images.githubusercontent.com/45469813/227717704-2aa4d8cb-516c-40aa-9bca-357b63e653d6.jpg)

**EDIT**
**2023-09-10: Added T version**
**2023-08-14: Added TR version** 
**2023-04-03: Revised description**

Wireless Tag has finally the new [WT32S3-86V](https://www.alibaba.com/product-detail/original-factory-86-box-screen-monitors_1600752497272.html?spm=a2700.shop_plgr.41413.3.4a4a12510gcShG) SKU ready for shipping 🎉
This SKU is a replacement for the deprecated [WT86-32-3ZW1](https://www.openhasp.com/0.7.0/hardware/wireless-tag/wt-86-32-3zw1/) SKU which has been unavailable since Q2, 2022
Sample device: 39.90 usd (programmer included) + shipping / tax
Still unfortunately only with Chinese sized endplate, which does not fit EU wall boxes
>**Note**
This new SKU is shipped as a 'developers version' and comes without PSU !!**
But device does come equipped with both speaker, microphones, audio- and RS485 chip populated on the pcb.

Device is 100% identical with the Panlee [ZX3D95CE01S-AR-4848](https://www.dropbox.com/s/g5z03ekfq4rtty9/ZX3D95CE01S-AR-4848.pdf?dl=0)
Outer casing and all dimensions also completely identical with the original WT86-32-3Z01 SKU

openHASP firmware binary is currently **not** available through [openHASP Nightly web builds](https://nightly.openhasp.com) but should be downloaded instead on [Github](https://github.com/HASwitchPlate/openHASP/actions)
**Actions -> Artifacts -> Panlee -> AR-4848 versions**
> **Note**
>THE PANLEE 4" DEVICE HAS BEEN RELEASED IN MULTIPLE VARIANTS WHERE THREE SKU'S AFAIK HAVE BEEN AVAILABLE THROUGH >ALIEXPRESS/ALIBABA !
>
>Currently only the latest released TR version is available.
>Could be that the AR- and UR versions are already discontinued and will remain unavailable.
>
>AR version [ZX3D95CE01S-AR-4848](https://www.dropbox.com/s/g5z03ekfq4rtty9/ZX3D95CE01S-AR-4848.pdf) which includes >speaker, microphones, audio- and RS485 chip.
>UR version  [ZX3D95CE01S-UR-4848](https://www.dropbox.com/s/y2hz152j6nlb74g/ZX3D95CE01S-UR-4848.pdf) which instead is equipped with an USB connector and the RS485 chip.
>T version which is equipped only with the RS485 chip. Can't find any active direct purchase links atm. 
>T version is identified as WT32S3-86S instead of the previous used WT32S3-86V
>TR version  [ZX3D95CE01S-TR-4848](https://www.dropbox.com/scl/fi/h08cineh0qz8wfh0i8mr5/ZX3D95CE01S-TR-4848.pdf?rlkey=lvyy9lnf176xb5xnvujhrtguj&dl=0) which instead is equipped with an temperature/humidity sensor (SHT20) and the RS485 chip.
>TR version is identified as WT32S3-86S instead of the previous used WT32S3-86V

Easily flashable using regular Dupont cables via the onboard 2mm pitch connector or the 2,54mm pitch flash header pins (unpopulated)
The [ZXACC-ESPDB-V20](https://vi.aliexpress.com/item/1005004271751158.html?spm=a2g0o.productlist.main.5.1a782f88GUixNP) FTDI serial programmer shipped with device, has both a 7 pin JST 1,25mm- as well as a 'standard' 2,54mm pitch pin connector. The 7 pin JST connector can be used to flash e.g. [WT32-SC01 Plus](https://www.openhasp.com/0.7.0/hardware/wireless-tag/wt32-sc01-plus/) devices via the debug connector
The 2mm pin connector is positioned on the pcb in exactly the same location as on the old SKU. Device is pin compatible for powering device with the old SKU backplate. But the relays unfortunately can't be operated, as the needed pins are not connected on the new SKU
A hardware hack is needed if you want to control the relays and some unused GPIO pins needs to be identified as well

Unless you need the microphones/speaker/RS485 interface, I would opt for the new [T3E v2](https://github.com/HASwitchPlate/openHASP/discussions/458) device instead. This device is equally priced, delivered with both PSU, three relays and will fit in a standard EU wallsocket. Both devices are delivered in same outer casing, contains the same high quality IPS display and have identical MCU specs.

**Cut from Wireless-Tag's Alibaba product description:**

- WT32-S3-WROVER-N16R8（16MB Flash/8Line 8M die inside)
- 480 * 480 high-resolution IPS RGB display with capacitive touch
- Everest Semiconductor’s ES8388 audio CODEC (Support dual microphone + speaker output)
**Note: Speaker/microphones currently unsupported in openHASP**

LCD Display: IPS
display size: 3.95inch
Resolution: 480X480
Flash memory: 128Mbit
Interface Type: RGB
Touch screen: Capacitive touch screen
Certification: FCC/CE-RED/SRRC
Mainboard power supply: 5V DC
Color: White
MCU: ESP32-S3-WROVER
Dimension: 86 x 86 x 37mm
Wi-Fi: 802.11 b/g/n

**Alibaba [link](https://www.alibaba.com/product-detail/original-factory-86-box-screen-monitors_1600752497272.html?spm=a2700.shop_plgr.41413.3.4a4a12510gcShG)**

![IMG20230325123446](https://user-images.githubusercontent.com/45469813/227718003-ea3b9f8c-c045-483c-aa3c-d08c6026e0a7.jpg)

![IMG20230325123558](https://user-images.githubusercontent.com/45469813/227717665-edfad9c7-95d0-455f-a785-f004498b6658.jpg)

**Comment by jazzmonger** [comment]
How is it different from the previous version?  Might the ghosting be fixed?

Chinese endplate means what exactly?  Will it fit in a N. American duplex box? 
I should start an auction for all the frigging displays I've bought in the past couple months... 🥵

**Reply by htvekov** [reply]
> How is it different from the previous version? Might the ghosting be fixed?
> 
> Chinese endplate means what exactly? Will it fit in a N. American duplex box? I should start an auction for all the frigging displays I've bought in the past couple months... 🥵

Well, hard to tell really, as I haven't recieved the device yet 😉
But my best guess is that the device housing is somewhat or completely identical to the original SKU.
If that's the case, then the actual part which goes into the wall box is appx. 68 x 68 x 25 mm. I've no idea about the N.A. wall box dimensions. I'm from Denmark  😆

The actual pcb and display will most likely be identical with the 4" Panlee device referenced above.
 Both specs for the display, driver chip, onboard RS485 and audio codec chip all match and is a strong indication that devices are identical. Meaning that display issues will be long gone, as the 480x480 res. IPS panel is far superior to the original SKU crappy display. 
 But time will tell...

**Comment by M4rv-dev** [comment]
Any info on how to use the RS485 on this board?

**Reply by fvanroie** [reply]
RS485 isn't supported by openHASP firmware...

**Comment by bensuffolk** [comment]
Look like this might be the same module on [AliExpress](https://vi.aliexpress.com/item/1005005188547984.html)

I'd be interested to know if it has spare I/O to add an SPI ethernet interface.

**Reply by htvekov** [reply]
Hard to tell if your AliExpress link is for the Panlee AR or UR version.
Buying from sellers crappy descriptions is pure guessing if they are not referring to a specific device model type.

If your not using the audio codec chip on the AR version or buying the UR version (without speakers, microphone and codec chip) at least GPIO 6 and 7 should be free and available.

Link for the UR version datasheet [here](https://www.dropbox.com/s/y2hz152j6nlb74g/ZX3D95CE01S-UR-4848.pdf?dl=0)
And the AR version datasheet [here](https://www.dropbox.com/s/g5z03ekfq4rtty9/ZX3D95CE01S-AR-4848.pdf?dl=0)

**Reply by bensuffolk** [reply]
Like you say, hard to tell with the generic descriptions. The photo shown as the USB connector, so in theory should be the UR version. Which would be fine by me as I have no interest in audio, as for which one would actually show up of course is anybodies guess as the photo could easily be a stock one and they might just randomly send whatever they have.

I have ordered one to put in my pile of 'that looks interesting, hopefully I'll get some time to play with it soon'. So I'll let you know which version actually does arrive.

**Reply by fvanroie** [reply]
It looks like that's the UR version, which is fine and supported. Just pick the right bin file and you're good.

![image](https://github.com/HASwitchPlate/openHASP/assets/15969459/fda2e8c1-53f6-4a17-95cf-b93b73f5db62)

**Reply by htvekov** [reply]
The LCD data line reference (8 bit 8080 mcu) in the AliExpress link seems strange ?
Guess it's just one more crappy Ali sales link with completely 'random' text, pictures and description 🙄

If you like the 4" square design, I would argue that the [GS-T3E](https://github.com/HASwitchPlate/openHASP/discussions/458) might be a better choice.
With the T3E you'll get a device with EU sized backplate, PSU and three relays.

The T3E has two (for openHASP unused) GPIO pins easily available at pads near the Tuya chip (U2)
Just desolder the Tuya chip or cut the power trace on the pcb.

The four relay GPIO pins are also easily accessible via the backplate connector (U13)

**Reply by hazymat** [reply]
Hi Ben - any luck getting one of these units sent over to you from AliEx? I'm also a UK buyer, I placed 3x orders on AE each with three units. One order automatically cancelled after a week, two of them were shipped, and one of those shipped orders magically unshipped itself (shipping status was "canceled") - didn't get refunds for either and had to contact AE on chat to request this, which itself is not so easy. Admittedly I bought the 86V versions with the plastic enclosed backbox as opposed to the one with metal frame that you referred to here. I think there are stock issues hence sellers just cancelling stuff. My aim is to end up with about 20 working models but I know just getting that far is going to be a bit of a difficulty given my experience so far. Interested to hear your experience. Thanks

**Reply by bensuffolk** [reply]
Yes it arrived the other day, its the UR version. No PSU or relays etc that you might get with other versions but that is no issue to me. I have not had any time to play with it except to plug it in and see it power up and run a test app thats installed on it.

**Comment by hazymat** [comment]
So it looks like the 86V has been discontinued recently. As of a week or two ago it's not possible to source it from Aliexpress (they are listed in stock with a number of different sellers and I placed orders, but they were all eventually cancelled).

So I placed an order from Alibaba for the newer 86S variant, and received it today. I bought it with the programmer.

I was sent [this data sheet](https://us-icbuim-file.oss-us-east-1.aliyuncs.com/file/2df90f4df0c48e6a637e21ec049fb2c7.pdf?Expires=1691044746&OSSAccessKeyId=LTAIxlzvmECyZvmX&Signature=dTkO83fjr7NnjhutB354z0YS%2FC8%3D&response-content-disposition=fileName%3DWT32S3-86S%EF%BC%88ZX3D95CE01S-TR-4848%EF%BC%89Datasheet-V1%201EN.pdf) from Wireless Tag, the manufacturer, although the image of the panel on that datasheet is not the same as the panel I received. I've requested / waiting for an updated version.

The programmer has this JST connector, and it was supplied with cable:
![image](https://github.com/HASwitchPlate/openHASP/assets/7063284/750a0e69-9c9b-4b1c-a3fd-1aa9e3611b71)

It was also shipped with a little JST PCB connector (SMD) with a square pad at each end, then 7 long pins, pictured below attached to the cable:
![image](https://github.com/HASwitchPlate/openHASP/assets/7063284/488b5437-9a20-4589-b059-02587759dc8a)

Rather than solder this JST connector onto the panel, looks like it would be possible to just hold it on there whilst programming (or use blue tac to hold it):

![image](https://github.com/HASwitchPlate/openHASP/assets/7063284/b53e5382-72e1-4643-838b-98d93403736b)

However when the USB-TTL board is plugged into my laptop's USB, although the USB-TTL board LED comes on and it's shown in device manager as a CH343, when I touch it onto the section of the panel marked "debug", no lights on the panel come on.

More testing required...

**Comment by htvekov** [comment]
Apparently a new TR version ZX3D95CE01S-TR-4848) released from Panlee. Never heard about this version until now.
Equipped with a SHT20 temp/humi sensor as a kind of 'break off board' on the pcb.

A quick glance at the TR datasheet reveals that it shares most pins with the [UR](https://www.dropbox.com/home/Public?preview=ZX3D95CE01S-UR-4848.pdf) version layout.
Seems that entire LCD pin layout is identical. I2C and touch pins needs to be revised though in order to get openHASP going on this device version.

Looks like Panlee are in a transition phase regarding this device.
Alibaba links now also reference to this new TR version. Text descriptions, images and video mostly reference to the 'old' UR and AR versions though (Audio codec chip still mentioned in text etc.)

Can anyone confirm that the old UR and AR versions both have been completely discontinued ?

**Reply by fvanroie** [reply]
The pin config for Arduino_GFX has been [posted here](https://github.com/moononournation/Arduino_GFX/discussions/329#discussioncomment-6692789).

**Reply by htvekov** [reply]
As usual the datasheet is somewhat fucked up... 🙄
Datasheet: BL pin 4
Schematics and Arduino_GFX code: BL pin 5

I'm a bit puzzled about the 'LCD autoreset' for the TFT_RST pin ? 🤔
Never seen that kind of hw config before. Guess pin should either be not defined or pin 41.
Arduino test code has the RST pin set to 41, so I guess that's the correct config then.

Datasheet/schematics and Arduino_GFX test code all combined ends up with this openHASP environment pin config for the TR version.
@hazymat Please check the Arduino_GFX test code @fvanroie posted on your TR device. If that checks out ok, hopefully this openHASP config should work as well. 

```
[env:panlee-zw3d95ce01s-tr-4848_16MB]
extends = panlee-zw3d95ce01s, flash_16mb
build_flags =
    -D HASP_MODEL="ZX3D95CE01S-TR-4848"
    ${panlee-zw3d95ce01s.build_flags}
    -D TOUCH_RST=-1 ; Using RC reset
    -D TOUCH_SCL=6  ; touch I2C bus clock, multiplexed with external interface
    -D TOUCH_SDA=15 ; Touch I2C bus data, multiplexed with external interface
    -D TOUCH_INT=4  ; Touch interrupt
    -D TFT_RST=41   ; LCD autoreset, use RGB_VS (TFT_VSYNC) signal to keep working
    -D TFT_DC=-1
    -D TFT_CS=38    ; LCD SPI bus CS
    -D TFT_SCLK=45  ; LCD SPI bus SCLK
    -D TFT_MOSI=48  ; LCD SPI bus MOSI
    -D TFT_MISO=-1 ; LCD SPI bus MISO
    -D TFT_DE=40    ; LCD RGB interface DE
    -D TFT_VSYNC=41    ; LCD RGB interface VS
    -D TFT_HSYNC=42    ; LCD RGB interface HS
    -D TFT_PCLK=39  ; LCD RGB interface PCLK
    -D TFT_B0=45    ; LCD RGB interface D0
    -D TFT_B1=48    ; LCD RGB interface D1
    -D TFT_B2=47    ; LCD RGB interface D2
    -D TFT_B3=0     ; LCD RGB interface D3
    -D TFT_B4=21    ; LCD RGB interface D4
    -D TFT_G0=14    ; LCD RGB interface D5
    -D TFT_G1=13    ; LCD RGB interface D6
    -D TFT_G2=12    ; LCD RGB interface D7
    -D TFT_G3=11    ; LCD RGB interface D8
    -D TFT_G4=16    ; LCD RGB interface D9
    -D TFT_G5=17   ; LCD RGB interface D10
    -D TFT_R0=18   ; LCD RGB interface D11
    -D TFT_R1=8    ; LCD RGB interface D12
    -D TFT_R2=3    ; LCD RGB interface D13
    -D TFT_R3=46   ; LCD RGB interface D14
    -D TFT_R4=10    ; LCD RGB interface D15
    -D TFT_BCKL=5   ; LCD backlight control, high level enable (datasheet states: 4, schematics states: 5 )  
```

**Reply by strange-v** [reply]
Here is a working test code for WT32S3-86S, Arduino_GFX, and LVGL:  https://github.com/strange-v/WT32S3-86S_LVGL
The panel doesn't work without specifying RST. Also, I wasn't able to use it with LovyanGFX.

**Reply by htvekov** [reply]
My openHASP env. config seems to match your test code as well 😎
Noticed that the hsync/vsync settings in openHASP ArduinoGFX build options differed slightly from the various Arduino_GFX test files scattered around.

Guess it's a typo somewhere. But not sure whether openHASP or the various Arduino_GFX test files actually has the right settings ?
I don't have a GC9503V datasheet with the correct settings at hand. 

```
openHASP - ArduinoGFX build options
-D TFT_HSYNC_FRONT_PORCH=10
-D TFT_HSYNC_PULSE_WIDTH=8
-D TFT_HSYNC_BACK_PORCH=50

-D TFT_VSYNC_FRONT_PORCH=10
-D TFT_VSYNC_PULSE_WIDTH=8
-D TFT_VSYNC_BACK_PORCH=20
```

**Reply by strange-v** [reply]
Theoretically, the correct value should be here: https://github.com/wireless-tag-com/ZX3D95CE01S-TR-4848/blob/7df1889589c8f0fb0343659707df2169ea867615/main/screen.c#L816C1-L821C36. But I don't have a datasheet either.

**Reply by htvekov** [reply]
At least Wireless-Tag has identical settings for both the AR- and TR versions in their repo's 🙂
IRL I don't believe that these minor variations has any significance. But I'm not really into LCD timing registers 😆 

```
.hsync_pulse_width = 10,
.hsync_back_porch = 40,
.hsync_front_porch = 8,
.vsync_pulse_width = 10,
.vsync_back_porch = 40,
.vsync_front_porch = 8,
```

**Reply by kinosang** [reply]
@htvekov works well with these definetion.

There actually has two variants (or 4) of the TR version:

- ZW3D95CE01S(-i)-TR with SHT20
- ZW3D95CE01S(-i)-R   without SHT20

**i** meaning IPEX antenna

Both of them have the same pin layout, ZW3D95CE01S-R just leaves the SHT20 pads empty.

![IMG_4253](https://github.com/HASwitchPlate/openHASP/assets/4987102/1a3ecc6d-db71-4b9d-9b45-ddddf076faef)
![IMG_4254](https://github.com/HASwitchPlate/openHASP/assets/4987102/99ee92d3-41fa-4bc0-a82e-ad376a6888fc)

<details>
<summary>Full configuration</summary>

```
[env:panlee-zw3d95ce01s-tr-4848_16MB]
extends = panlee-zw3d95ce01s, flash_16mb
build_flags =
    -D HASP_MODEL="ZX3D95CE01S-TR-4848"
    ${panlee-zw3d95ce01s.build_flags}
    -D TOUCH_RST=-1 ; Using RC reset
    -D TOUCH_SCL=6  ; touch I2C bus clock, multiplexed with external interface
    -D TOUCH_SDA=15 ; Touch I2C bus data, multiplexed with external interface
    -D TOUCH_INT=4  ; Touch interrupt
    -D TFT_RST=41   ; LCD autoreset, use RGB_VS (TFT_VSYNC) signal to keep working
    -D TFT_DC=-1
    -D TFT_CS=38    ; LCD SPI bus CS
    -D TFT_SCLK=45  ; LCD SPI bus SCLK
    -D TFT_MOSI=48  ; LCD SPI bus MOSI
    -D TFT_MISO=-1 ; LCD SPI bus MISO
    -D TFT_DE=40    ; LCD RGB interface DE
    -D TFT_VSYNC=41    ; LCD RGB interface VS
    -D TFT_HSYNC=42    ; LCD RGB interface HS
    -D TFT_PCLK=39  ; LCD RGB interface PCLK
    -D TFT_B0=45    ; LCD RGB interface D0
    -D TFT_B1=48    ; LCD RGB interface D1
    -D TFT_B2=47    ; LCD RGB interface D2
    -D TFT_B3=0     ; LCD RGB interface D3
    -D TFT_B4=21    ; LCD RGB interface D4
    -D TFT_G0=14    ; LCD RGB interface D5
    -D TFT_G1=13    ; LCD RGB interface D6
    -D TFT_G2=12    ; LCD RGB interface D7
    -D TFT_G3=11    ; LCD RGB interface D8
    -D TFT_G4=16    ; LCD RGB interface D9
    -D TFT_G5=17   ; LCD RGB interface D10
    -D TFT_R0=18   ; LCD RGB interface D11
    -D TFT_R1=8    ; LCD RGB interface D12
    -D TFT_R2=3    ; LCD RGB interface D13
    -D TFT_R3=46   ; LCD RGB interface D14
    -D TFT_R4=10    ; LCD RGB interface D15
    -D TFT_BCKL=5   ; LCD backlight control, high level enable (datasheet states: 4, schematics states: 5 )
    -D TFT_HSYNC_PULSE_WIDTH=10
    -D TFT_HSYNC_BACK_PORCH=40
    -D TFT_HSYNC_FRONT_PORCH=8
    -D TFT_VSYNC_PULSE_WIDTH=10
    -D TFT_VSYNC_BACK_PORCH=40
    -D TFT_VSYNC_FRONT_PORCH=10
```

</details>

**Reply by htvekov** [reply]
@fvanroie 
Should we add the TR environment now build options has been confirmed working ?

**Reply by fvanroie** [reply]
yes sure, it's a valid SKU now

**Reply by htvekov** [reply]
OK. I'm afraid that I don't know what else is needed besides the build environment to do a PR.
Did you notice my comment on hsync/vsync settings. An initial typo or... ?

**Reply by htvekov** [reply]
I know about the T & TR variants.
As openHASP doesn't natively support the temp. sensor (at least not at the moment) both variants can share identical build env.

Don't really understand your note regarding the two/four variants. Internal pcb antenna / external IPEX connector antenna or... ?

EDIT: OK, found the IPEX versions mentioned in the datasheet. External IPEX connector version possible for batch orders.
![image](https://github.com/HASwitchPlate/openHASP/assets/45469813/95491037-e5d8-4b19-bc06-1ae683f79c4d)

**Reply by kinosang** [reply]
I wrote these notes just to remind others, cuz I confused -R and -AR/-UR at the beginning.

I didn't test other hsync/vsync settings, just use the values as-is.

**Comment by AdrienV** [comment]
> openHASP firmware binary is currently not available through [openHASP Nightly web builds](https://nightly.openhasp.com/) but should be downloaded instead on [Github](https://github.com/HASwitchPlate/openHASP/actions)
Actions -> Artifacts -> Panlee -> AR-4848 versions

Hi guys,
@htvekov sorry but I don't see "Artifacts" in Action menu, could you tell me where to find the last firmware please ?

**Reply by htvekov** [reply]
You first click on the top `Actions` menu. Then select and click on whatever `workflow run` you want to download (usually the top one) and after that the `Artifacts` are shown.
Find `Panlee` and click on it. Then you'll download a zip file containing full and ota binaries for Panlee `AR`, `UR` and `TR` versions  

![image](https://github.com/HASwitchPlate/openHASP/assets/45469813/06fe694b-d6f9-4d8b-a4f2-567695ce267e)

**Reply by AdrienV** [reply]
Thanks so much :)

**Comment by htvekov** [comment]
![image](https://github.com/HASwitchPlate/openHASP/assets/45469813/c70d7208-3913-4afa-9767-183e4925a5a5)

**Comment by AdrienV** [comment]
Thanks again @htvekov, I write the steps I followed if it can help newbies like me to load HASP on a Panlee WT32-S3-WROVER.

### 1 - Install [VSCODE Espressif extension](https://github.com/espressif/vscode-esp-idf-extension/blob/master/docs/tutorial/install.md) 
 // or follow [these steps](https://docs.espressif.com/projects/esp-idf/en/latest/esp32s3/get-started/index.html) for other installation methods

### 2 - Connect your Panlee with [CH340 module](https://fr.aliexpress.com/item/1005005140380360.html)  like this : 
_If you have a switch 3V3 or 5V on your CH340G, choose 3V3_

<img width="912" alt="Capture d’écran 2023-09-18 à 22 42 02" src="https://github.com/HASwitchPlate/openHASP/assets/3740879/08125826-785a-4ff6-8af7-48231d9fc5fd">

**PANLEE  |||  CH340 / CH340G**
+5V   |||   5V
GND   |||   GND
TX0   |||   RXD
RX0   |||   TXD

**PANLEE  |||  PANLEE**
ID0   |||   GND

### 3 - Download a firmware ###

Follow these steps (thanks @htvekov)
> You first click on the top `Actions` menu. Then select and click on whatever `workflow run` you want to download (usually the top one) and after that the `Artifacts` are shown. Find `Panlee` and click on it. Then you'll download a zip file containing full and ota binaries for Panlee `AR`, `UR` and `TR` versions
> 
> ![image](https://user-images.githubusercontent.com/45469813/268757435-06fe694b-d6f9-4d8b-a4f2-567695ce267e.png)

### 4 - Flash your Panlee ###
- Check on which port is connected your Panlee (ls /dev/cu.*  on Mac, Go to "Device manager" on Windows & check "Ports (COM & LPT) to find your Panlee port)

- Open a VSCode Terminal and **ERASE YOUR DEVICE** : 

`esptool.py --port /dev/xxx erase_flash ` (/dev/xxx must be replaced by your Panlee port)

- Write the firmware downloaded at step 3 on your Panlee : 

`esptool.py --port /dev/xxx write_flash 0x0 '/path-to-your-firmware/panlee-zw3d95ce01s-ar-4848_ota_v0.7.0-rc8_f28627c.bin'` (/path-to-your-firmware/panlee-zw3d95ce01s-ar-4848_ota_v0.7.0-rc8_f28627c.bin must be replaced by your choosed firmware



I hope this can help others :) If there is mistake in my steps or terms used don't hesitate to tell me...

**Reply by htvekov** [reply]
When doing serial flash you need to use the `full` binary version!!
The `ota` files are for WiFi 'Over The Air' updates. These can be used **_after_** the initial openHASP fw serial flash has been done.

**Comment by smcgann99** [comment]
@htvekov I was looking for the NEW T3E on alliexpress and found this - https://www.aliexpress.com/item/1005005906705801.html?spm=a2g0o.cart.0.0.2a7c38daZz9JcR&mp=1
Is there anyway to tell if it's the same device ?

**Reply by htvekov** [reply]
Well, the text description refers to EU backplate and relays. So AFAIK it really can't be anything else than a [T3E](https://github.com/HASwitchPlate/openHASP/discussions/458) device. Just make sure that you get the pcb revision v2.3 which is the latest version. Don't expect a reseller to know anything about the device pcb revision.

I've bought all my T3E directly from the producer Golden Security  via [Alibaba](https://www.alibaba.com/product-detail/2022-new-arrival-smart-wifi-switch_1600573806214.html?spm=a2700.details.0.0.252d69cf8GeWdp)
Price is appx. 40 usd + shipping costs. Depending on the quantity you order, it's most likely cheaper to buy there than via your Aliexpress link.

**Reply by smcgann99** [reply]
**Don't expect a reseller to know anything about the device pcb revision.**
There is the problem - I have asked that question but no response ;-)
Never used Alibaba as they tend to have large min order qty and unknown shipping costs. 
I might see if I can specify version with order then I can always send it back as not matching description ?

**Reply by htvekov** [reply]
I've ordered T3E devices three times from Golden Security. No issues at all. You can order low quantities as well (even 1 pcs. as test sample)
Just ask Maggie at the chat to ensure you get the pcb 2.3 version
Shouldn't be an issue since the old pcb v1.x has been deprecated for many months now.

**Reply by cook1emr** [reply]
Just to chime in, I have also ordered from this Alibaba seller and had no issues either.

I am UK like you @smcgann99

**Reply by nagyrobi** [reply]
Aliexpress would be preferable as already has taxes/vat included in the price so shipping is less cumbersome.

**Reply by htvekov** [reply]
Alibaba also provides the shipping cost and tax (if you don't have a VAT number) at check out.
But I agree that it's easier to compare prices from Aliexpress as prices shown are including tax/vat

![image](https://github.com/HASwitchPlate/openHASP/assets/45469813/d56922d5-643e-41d3-af55-9af475c51f5a)

**Comment by PARitter** [comment]
Does the Panlee "TR" build happen to expose the SHT20 Temp/Humidity data (longshot, I know, since I2C based sensors are not generally supported by OpenHASP).

**Reply by htvekov** [reply]
Nope. You'll have to do your own Custom Code for the SHT20

**Comment by smcgann99** [comment]
There seem to be a lot of these TR with sht20 around on Aliexpress etc. I have looked at the custom options but I'm afraid its a bit beyond my knowledge at the moment. 
Is there any chance this could be added to the "standard" panlee-zw3d95ce01s-tr-4848_16MB build for this model ?

**Comment by smcgann99** [comment]
Has anyone already done a custom build for the onboard SHT20, if so could you provide a guide please?

**Comment by KristianDebrenti** [comment]
Hello, I have compiled a custom build for **panlee-zw3d95ce01s-tr-4848_16MB** adding support for SHT20 sensor and it is available here: **https://github.com/KristianDebrenti/openHASP**.
When installing for the first time, use the "full" bin file, then you may update it with a newer release using the OTA bins.
The SHT20 Temperature & Humidity values are being sent every 5 seconds via MQTT on the topic **hasp/plate/state/sht20** as JSON object.
Due to the fact that the SHT20 sensor contained in the same enclosure with the ESP32 and display controller etc, you need to apply some sort of manual adjustment to the returned values. In my experience, I found that I achieve usable-ish results by substracting ~9 from the temperature value and add ~50 to the humidity.

**Reply by smcgann99** [reply]
Thanks for sharing, I will give it a try. 
Based on your experience it seems like it's not very accurate then ;-)
Also shouldn't the value be in the existing sensor topic, and use the Tele Period set in debug ?

**Reply by KristianDebrenti** [reply]
Hi, regarding _"Also shouldn't the value be in the existing sensor topic, and use the Tele Period set in debug ?"_ - I don't know how to do this, but I am open to learn if you give me a hint. Compiling this custom build was my first ever such experience and first time using Visual Studio Code as I am not a developer, just a IoT passionate 😊

**Reply by smcgann99** [reply]
You are at least 1 step ahead of me then ;-)

Perhaps someone else can assist you ?

**Reply by tonyis49** [reply]
Hello, great work. I have had the panel working for the last few days and been experimenting with different layouts. I have just uploaded your latest build and I can see the temperature and humidity values showing in the MQTT Explorer. As a relative newbie I have tried to incorporate these values into my mqtt.yaml but not having any luck reading the values.
    # Panlee Temperature
  - name: "Panlee Temperature"
    unique_id: "panlee_temperature"
    state_topic: "hasp/plate01/state/sht20"
    value_template: '{{ value_json.value|float(0)|round(1) }}'
    unit_of_measurement: "°C"   

maybe you could help?

**Reply by tonyis49** [reply]
worked it out with this and works
    # Panlee Temperature
  - name: "Panlee Temperature"
    unique_id: "panlee_temperature"
    state_topic: "hasp/plate01/state/sht20"
    value_template: '{{ value_json.Temperature|float(0)|round(1) }}'
    unit_of_measurement: "°C"

    # Panlee Humidity
  - name: "Panlee Humidity"
    unique_id: "panlee_humidity"
    state_topic: "hasp/plate01/state/sht20"
    value_template: '{{ value_json.Humidity|float(0)|round(1) }}'
    unit_of_measurement: "%"

**Reply by tonyis49** [reply]
> Hello, I have compiled a custom build for **panlee-zw3d95ce01s-tr-4848_16MB** adding support for SHT20 sensor and it is available here: **https://github.com/KristianDebrenti/openHASP**. When installing for the first time, use the "full" bin file, then you may update it with a newer release using the OTA bins. The SHT20 Temperature & Humidity values are being sent every 5 seconds via MQTT on the topic **hasp/plate/state/sht20** as JSON object. Due to the fact that the SHT20 sensor contained in the same enclosure with the ESP32 and display controller etc, you need to apply some sort of manual adjustment to the returned values. In my experience, I found that I achieve usable-ish results by substracting ~9 from the temperature value and add ~50 to the humidity.

Hi, I don't suppose you could share the source code for me to include in my own custom build.

**Comment by KristianDebrenti** [comment]
> Thanks for sharing, I will give it a try. Based on your experience it seems like it's not very accurate then ;-) Also shouldn't the value be in the existing sensor topic, and use the Tele Period set in debug ?

Just a glimpse at the built in sensor accuracy - adjusted SHT20 values (red) vs. Shelly HT (blue):
![image](https://github.com/HASwitchPlate/openHASP/assets/48323301/8e70026d-f634-4cb9-a3d5-4d1c211ac7f4)

**Reply by KristianDebrenti** [reply]
I have purchased a second panel, identical, from the same provider and the offset adjustment of the temperature and humidity values are different for this second screen. It seems like you need to adjust the offset individually for each particular screen.

**Reply by smcgann99** [reply]
Humm, seems the sensors are not much use then.

**Comment by ott** [comment]
It seems that are at least three sellers on Aliexpress that sell the ZX3D95CE01S-TR together with a power supply and RS-485 transmit and receive screw terminals. The devices are designated as “ZXD95CE01S-TR Power”.  The brand on the power supply PCB is “QSZNTEC”. 

The devices cost more than comparable devices but have a RS-485 interface and an included power supply. A power supply that fits inside an a cavity wall connection box is also not inexpensive. So I think the higher price is justified. I ordered two devices and I'm going to publish photos of the power supply here when it has arrived.

It seems that the power supply does not have a safety capacitor populated. [This has also been reported for a similar power supply for the Guition ESP32-S3-4848S040](https://michiel.vanderwulp.be/domotica/Modules/SmartDisplay-ESP32-S3-4.0inch/power-supply/). I will have a look at the datasheet of the power supply module on the power supply PCB whether it is required or recommended but it does not seem to be a good sign.

The RS-485 interface is also strange. The power supply PCB only offers transmit and receive screw terminals but no ground. The RS-485 transceiver is a SP3485E (a clone of the MaxLinear SP3485). So the RS-485 interface is also not isolated. While it seems possible to design RS-485 buses without a common ground through some form of isolation amplifier and some RS-485 isolation and repeater devices that are designed for long distance transmission seem to make use of such ICs, at least the Modbus specification requires a common ground. The RS-485 transceiver on the device is also not designed for high common-mode voltages and, therefore, ground potential differences. It seems that the designers did not consider this or assumed that the ground-potential differences are within the voltage range of the transceiver.

Given that the device also uses two 10 kΩ resistors for fail-safe biasing with 3.3 V according to the schematics, I have some doubts about how much thought went into the design. However, it would still be possible to solder a wire to ground of secondary side of the power supply and pass it out of the enclosure through the vents or an additional hole. The resistors can also be changed or omitted. So it seems that this mistake would be correctable (even though such modification would likely never pass certification for valid reasons). Another option would be to use a RS-485 isolator that does not seem to require ground to measure the differential pair. It seems that the Advantech ADAM-4510 or perhaps the Kemei M-11C, which claims to use Analog Devices iCoupler technology, might be able to handle this. RS-485 seems to be a mess in this regard and the manufacturer of this device is not the first to offer RS-485 in this way. So I'm hopeful that there is some solution.

I'm not an expert on RS-485 and more complicated electrical engineering questions but I will nonetheless also report what I could find out about RS-485 on the device when it arrives.

I also ordered an [Europa ECWSK1](https://www.rexel.co.uk/uki/root-category/Switches%2C-Sockets-%26-Wiring/Switches-%26-Sockets/Weatherproof-Outdoor/Enclosure-Insulated-1-Gang-Outdoor-Socket-IP65-110X140X80mm-Abs/p/2500362478) socket enclosure to install the device into for an (roof covered) outdoor installation.

![Sb12a8a6860b346a89ff7c73b1746f451O](https://github.com/user-attachments/assets/a89c2cc2-c0dd-44cf-bc65-d8f5ffe26c20)
